namespace/namespace.go

```diff
@@ -23,10 +22,14 @@ func New(version uint8, id []byte) (Namespace, error) {
 		return Namespace{}, err
 	}
 
+	return newNamespace(version, id), nil
+}
+
+func newNamespace(version uint8, id []byte) Namespace {
+	data := []byte{version}
 	return Namespace{
-		Version: version,
-		ID:      id,
-	}, nil
+		data: append(data, id...),
+	}
```

Comment:
Allocation police 🚓 
This allocates twice, but we could do it in a single allocation. One of the way to do that is similar to https://github.com/celestiaorg/celestia-node/blob/540192259c144ccbd24e45e34616a41389232a51/share/namespace.go#L46-L50